File: PHP/EZTV.php
<?php
    include '/Helpers.php';
    class EZTV
    {
        static public function search($query)
        {
            $results = [];
            $url = "https://eztv.ag/search/".urlencode($query);
            libxml_use_internal_errors(true);
            $html = preg_replace('/>\s+</', '><', curl($url));
            $dom = DOMDocument::loadHTML($html);
            $nodelist = DOMXPath($dom)->query('//tr[contains(@name, "hover")]');
            foreach ($nodelist as $node) {
                $results[] = array(
                    'name' => $node->childNodes->item(1)->firstChild->nodeValue,
                    'size' => $node->childNodes->item(3)->nodeValue,
                    'seeds' => $node->childNodes->item(5)->firstChild->nodeValue,
                    'peers' => "-",
                    'link' => "https://eztv.ag/".$node->childNodes->item(1)->firstChild->getAttribute('href'),
                    'magnet' => $ref = $node->childNodes->item(2)->firstChild->getAttribute('href'),
                    'hash' => preg_replace('/^.+btih:([A-Za-z0-9]+).*/', '$1', $ref)
                );
            }
            return $results;
        }
    }

File: Python/eztv.py
from lxml import html
import requests
from helpers import filtor

def search(q):
    url = f'https://eztv.ag/search/{q}'
    data = requests.get(url)
    tree = html.fromstring(data.content)
    results = zip(
        tree.xpath('//tr[@name="hover"]//a[@class="epinfo"]/@title'),
        tree.xpath('//tr[@name="hover"]/td[4]/text()'),
        tree.xpath('//tr[@name="hover"]/td[6]//text()'),
        tree.xpath('//tr[@name="hover"]/td[3]/a[1]/@href'),
        tree.xpath('//tr[@name="hover"]/td[1]/a/@title')
    )
    torrents = []
    for name, size, seed, mag, show in results:
        if mag.startswith('magnet') and 'Other' not in show:
            name = name.replace(f' ({size})', '')
            seed = seed.replace(',', '')
            if filtor(name, seed, q):
                torrents.append({
                    'name': name,
                    'size': size,
                    'seeds': seed,
                    'peers': '-',
                    'magnet': mag,
                    'hash': mag[20:60]
                })
    return torrents

print(search("the+librarians"))
